src/main/java/rx/schedulers/ScheduleWhen.java

```diff
@@ -0,0 +1,263 @@
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.schedulers;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import rx.Completable;
+import rx.Completable.CompletableOnSubscribe;
+import rx.Completable.CompletableSubscriber;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.annotations.Experimental;
+import rx.functions.Action0;
+import rx.functions.Func1;
+import rx.internal.operators.BufferUntilSubscriber;
+import rx.subjects.PublishSubject;
+import rx.subscriptions.Subscriptions;
+
+/**
+ * Allows the use of operators for controlling the timing around when actions
+ * scheduled on workers are actually done. This makes it possible to layer
+ * additional behavior on an existing {@link Scheduler}. The main parameter is a
+ * function that flattens an {@link Observable} of {@link Completable}s into
+ * just one {@link Completable}. There must be a chain of operators connecting
+ * the returned value to the source {@link Observable} otherwise any work
+ * scheduled on the {@link Scheduler} will not be executed.
+ * <p>
+ * When {@link Scheduler#createWorker()} is invoked a {@link Completable} is
+ * onNext'd to the combinator to be flattened. If the {@link Completable} is not
+ * immediately subscribed to an calls to {@link Worker#schedule} are buffered.
+ * Once the {@link Completable} is subscribed to actions are then scheduled on the
+ * actual {@link Scheduler}. When the {@link Worker} is unsubscribed the
+ * {@link Completable} emits an onComplete and triggers any behavior in the
+ * flattening operator. The {@link Observable} and all {@link Completable}s give
+ * to the flattening function never onError.
+ * <p>
+ * Slowing down the rate to no more than than 1 a second.
+ * 
+ * <pre>
+ * Scheduler slowSched = new ScheduleWhen(workers -> {
+ * 	// use concatenate to make each worker happen one at a time.
+ * 	return Completable.concat(workers.map(worker -> {
+ * 		// delay the starting of the next worker by 1 second.
+ * 		return worker.delay(1, TimeUnit.SECONDS);
+ * 	}));
+ * }, Schedulers.computation());
+ * </pre>
+ * <p>
+ * Limit the amount concurrency two at a time without creating a new fix size
+ * thread pool:
+ * 
+ * <pre>
+ * Scheduler limitSched = new ScheduleWhen(workers -> {
+ * 	// use merge max concurrent to two at a time
+ * 	return Completable.merge(workers, 2);
+ * }, Schedulers.computation());
+ * </pre>
+ */
+@Experimental
+public class ScheduleWhen extends Scheduler {
```

Comment:
The name was from following the theme of `retryWhen` and `repeatWhen` that use `Observable` as a control flow and not a data flow.